mpsl/Kconfig

```diff
@@ -10,6 +10,7 @@ config MPSL
 	select ZERO_LATENCY_IRQS
 	select IEEE802154_NRF5_EXT_IRQ_MGMT if IEEE802154_NRF5
 	select NRF_802154_MULTIPROTOCOL_SUPPORT if NRF_802154_RADIO_DRIVER
+	select MULTITHREADING_LOCK
```

Comment:
@tejlmand , please split this out to a separate commit

Reply:
To my knowledge, changes required because of an upmerge should go together, otherwise bisectability will be broken.
Two commits will mean that builds are broken in between.

Reply:
commit message updated.

Reply:
@tejlmand I think the update to the manifest is what counts as it will pull all the new changes in. So it can be two commits. Maybe @carlescufi can elaborate?

Reply:
No, it has to be a single commit if we are to keep this bisectable @rugeGerritsen and @joerchan.